operator/controllers/conditions.go

```diff
@@ -0,0 +1,91 @@
+package controllers
+
+import (
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Reason = string
+
+const (
+	ReconcileSucceeded     Reason = "ReconcileSucceeded"
+	Initialized            Reason = "Initialized"
+	Processing             Reason = "Processing"
+	OlderCRExists          Reason = "OlderCRExists"
+	ChartInstallFailed     Reason = "ChartInstallFailed"
+	ConsistencyCheckFailed Reason = "ConsistencyCheckFailed"
+	MissingSecret          Reason = "MissingSecret"
+	InvalidSecret          Reason = "InvalidSecret"
+	HardDeleting           Reason = "HardDeleting"
+	ResourceRemovalFailed  Reason = "ResourceRemovalFailed"
+	HardDeleteFailed       Reason = "HardDeleteFailed"
+	SoftDeleteFailed       Reason = "SoftDeleteFailed"
+	SoftDeleting           Reason = "SoftDeleting"
+	Recovered              Reason = "Recovered"
+)
+
+type TypeAndStatus struct {
+	Status metav1.ConditionStatus
+	Type   string
+}
+
+var Ready = TypeAndStatus{
+	Status: metav1.ConditionTrue,
+	Type:   "Ready",
+}
+
+var NotReady = TypeAndStatus{
+	Status: metav1.ConditionFalse,
+	Type:   "Ready",
+}
+
+var Reasons = map[Reason]TypeAndStatus{
+	ReconcileSucceeded:     Ready,
+	Initialized:            Ready,
+	Recovered:              Ready,
+	ChartInstallFailed:     NotReady,
+	ConsistencyCheckFailed: NotReady,
+	Processing:             NotReady,
+	OlderCRExists:          NotReady,
+	MissingSecret:          NotReady,
+	InvalidSecret:          NotReady,
+	HardDeleting:           NotReady,
+	ResourceRemovalFailed:  NotReady,
+	HardDeleteFailed:       NotReady,
+	SoftDeleteFailed:       NotReady,
+	SoftDeleting:           NotReady,
+}
+
+func NewConditionForReason(reason string, message string) *metav1.Condition {
```

Comment:
Method name `NewConditionForReason` suggests creation of a new Condition from passed arguments. How about changing the signature of the method to something like `ConditionFromExistingReason(reason Reason, message string)`?

Reply:
Done.